LandSandBoat/server
# Refactor Gear Sets

**_I affirm:_**

## What does this pull request do?
Refactors Gear Sets for the following items:
* Reformats and Expands gearSets table
* All Item IDs are now using items.lua enum
* Mod Values are based on number of set pieces equipped for a set (no math, just N set pieces starting at minEquipped)
* Dynamically creates Item ID lookup to set ID table on cache to prevent potentially iterating over all sets/items
* Updates logic for Lua bindings, removes unused hasGearSetMod binding
* Sets maximum limit for equipped items in set (optional)
* Audits and updates previously defined set definitions



## Steps to test these changes
Equip set and non-set items, see mods applied and removed as expected.

![image](https://user-images.githubusercontent.com/81713309/191310206-c55fd333-9f2b-4a9b-ac77-1dd0fa35f264.png)
